### PR TITLE
Upgrade jest

### PR DESCRIPTION
**Pull request by uladzimirdev**
### Description

Remember, when you needed to wait for 1+ min to run just a single test? With this single upgrade your experience should be improved dramatically.

| Before | After |
|--------|--------|
| Whole test suite with 4 workers and no cache on m2max  | `yarn jest --maxWorkers=4 --config jest.unit.conf.json --no-cache --coverage=false` |
| image | image |
| Single test with no cache on m2max |
| image | image |
| https://www.loom.com/share/a9de0748f9794a808aca184551625941 | https://www.loom.com/share/33b72b1d147b4273aa2e011e0e215391 | 

- - -
This change is [Reviewable](https://reviewable.io/reviews/metabase/metabase/30860)

### CODE REVIEW

File: frontend/src/metabase/components/FieldValuesWidget/FieldValuesWidget.unit.spec.js [hunk 1]
```diff
@@ -160,7 +160,6 @@ describe("FieldValuesWidget", () => {
       it("should not call fetchFieldValues", () => {
         const { fetchFieldValues } = renderFieldValuesWidget({
           fields: [field],
-          fetchFieldValues,
```
Comment (uladzimirdev) [inline frontend/src/metabase/components/FieldValuesWidget/FieldValuesWidget.unit.spec.js:163]: I think it was passed by mistake

File: package.json [hunk 2]
```diff
@@ -234,9 +234,9 @@
     "html-webpack-harddisk-plugin": "^2.0.0",
     "html-webpack-plugin": "5.5.0",
     "husky": "^8.0.2",
-    "jest": "^29.3.1",
+    "jest": "^29.5.0",
```
Comment (uladzimirdev) [inline package.json:237]: main change

**Review by ranquild — APPROVED** [review]

File: .github/workflows/frontend.yml [hunk 3]
```diff
@@ -81,6 +81,11 @@ jobs:
     runs-on: buildjet-2vcpu-ubuntu-2204
     timeout-minutes: 30
     steps:
+    - name: Get number of CPU cores
```
Comment (uladzimirdev) [inline .github/workflows/frontend.yml:84]: I'll revert it, just for testing

File: frontend/src/metabase/components/FieldValuesWidget/FieldValuesWidget.unit.spec.js (hunk 1 above)
Comment (nemanjaglumac) [reply frontend/src/metabase/components/FieldValuesWidget/FieldValuesWidget.unit.spec.js:163]: Was this breaking this test?

Comment (uladzimirdev) [reply frontend/src/metabase/components/FieldValuesWidget/FieldValuesWidget.unit.spec.js:163]: yes, the value is used before it's defined with `const`

**Review by nemanjaglumac — APPROVED** [review]
Awesome.
If you get all tests passing, ship it! 🚢

**Pull request merged**